### ISSUE TITLE
Missing GLES support

### ISSUE THREAD

**Issue by podsvirov** [coin3d/coin#449]
Strategically, this is a step towards supporting [WebAssembly](https://webassembly.org/) via [Emscripten](https://emscripten.org/) and creating 3D viewers in modern browsers.

PS: If I'm wrong about something, then correct me.

**Comment by VolkerEnderlein** [comment]
No, you are definitely right. Besides implementing the EGL support for Wayland this is another important issue that needs to be done to keep Coin alive and kicking :)

**Comment by luzpaz** [comment]
We have a showstopper here.

**Comment by podsvirov** [comment]
I have experience to [port](https://github.com/danginsburg/opengles3-book/pull/69) `OpenGL ES 3.0 Programming Guide Sample Code` to `Emscripten`. It uses both `EGL` and `GLES` API. But I don't have enough free time and knowledge of `Coin` architecture to apply this experience.

**Comment by VolkerEnderlein** [comment]
Unfortunately my OpenGL knowledge is very basic. I've had a look into the Coin wgl stuff but could not derive a proper schedule what needs to be done to implement egl bindings. That's why I marked this as `help wanted`. Sorry.

**Comment by luzpaz** [comment]
Anyone know of someone that could tackle this? If not, how to we proceed?

**Comment by luzpaz** [comment]
CC @wmayer @yorikvanhavre @realthunder @kkremitzki

**Comment by VolkerEnderlein** [comment]
@epuzanov has been kindly working on this and provided PR #470 for this. I have tested this and just merged with master.

**Comment by hfiguiere** [comment]
I built FreeCAD (as a flatpak) with coin3d from `master` (commit 28ac68d3de1970b32fd9524ad1ba6d0a502b78f2), passing `-DCOIN_BUILD_EGL=YES`. On x86_64, no problem. 

But on aarch64 (it's built with a Qt5 that only has GLES, and it's running on a RaspberryPi 4 64-bits), when I open a model I get this errors:

```
Coin warning in cc_glglue_instance(): Error when setting up the GL context. This can happen if there is no current context, or if the context has been set up incorrectly.
Coin warning in SoGLPolygonOffsetElement::updategl(): OpenGL driver doesn't support z-buffer offsetting
```

And then the display looks like this:
![freecad](https://user-images.githubusercontent.com/114441/148788261-57ca9929-4b39-494c-8cea-a85a30d0e2d1.png)

**Comment by epuzanov** [comment]
The GLES is not an EGL. The Coin3d library used an OpenGL API, some part of it must be rewritten to be compatible with OpenGL ES API. The EGL support is needed for compatibility with Wayland, because Wayland doesn't support GLX API.
Summary:
GLX (X11) -> EGL (Wayland)
OpenGL -> OpenGL ES (GLES)

**Comment by hfiguiere** [comment]
Which I is I thought I had understood, but https://github.com/coin3d/coin/issues/449#issuecomment-1007506992 threw me off thinking it would work.

At least it no longer crashes, but that could be a different reason.

(yes the RaspberryPi still doesn't use Wayland/Weston)

**Comment by VolkerEnderlein** [comment]
@hfiguiere So it seems to be my fault. I thought it is something conntected to the EGL work. Sorry for the noise.

**Comment by luzpaz** [comment]
Any progress on this?

**Comment by luzpaz** [comment]
bump
